source/adapter.screenshare.js

```diff
@@ -0,0 +1,165 @@
+          if (error.name === 'PermissionDeniedError' && window.parent.location.protocol === 'https:') {
+            console.error(error);
+            window.location.href = 'https://cdn.temasys.com.sg/skylink/extensions/skylink-webrtc-tools.xpi';
+          } else {
+            failureCb(error);
+          }
+        })
+
+      } else { // regular GetUserMediaRequest
+        baseGetUserMedia(constraints, successCb, failureCb);
+      }
+    };
+
+    window.getUserMedia = window.navigator.getUserMedia;
+
+  } else if (window.navigator.webkitGetUserMedia) {
+    baseGetUserMedia = window.navigator.getUserMedia;
+
+    window.navigator.getUserMedia = function (constraints, successCb, failureCb) {
+      constraints = constraints || {};
+
+      if (constraints.video && !!constraints.video.mediaSource) {
+        if (window.webrtcDetectedBrowser !== 'chrome') {
+          throw new Error('Current browser does not support screensharing');
+        }
+
+        var chromeCallback = function(error, sourceId) {
+          if(!error) {
+            constraints.video.mandatory = constraints.video.mandatory || {};
+            constraints.video.mandatory.chromeMediaSource = 'desktop';
+            constraints.video.mandatory.maxWidth = window.screen.width > 1920 ? window.screen.width : 1920;
+            constraints.video.mandatory.maxHeight = window.screen.height > 1080 ? window.screen.height : 1080;
+
+            if (sourceId) {
+              constraints.video.mandatory.chromeMediaSourceId = sourceId;
+            }
+
+            delete constraints.video.mediaSource;
+
+            baseGetUserMedia(constraints, successCb, failureCb);
+
+          } else {
+            if (error === 'permission-denied') {
+              throw new Error('Permission denied for screen retrieval');
+            } else {
+              throw new Error('Failed retrieving selected screen');
+            }
+          }
+        };
+
+        var onIFrameCallback = function (event) {
+          if (!event.data) {
+            return;
+          }
+
+          if (event.data.chromeMediaSourceId) {
+            if (event.data.chromeMediaSourceId === 'PermissionDeniedError') {
+                chromeCallback('permission-denied');
+            } else {
+              chromeCallback(null, event.data.chromeMediaSourceId);
+            }
+          }
+
+          if (event.data.chromeExtensionStatus) {
+            chromeCallback(event.data.chromeExtensionStatus, null);
+          }
+
+          // this event listener is no more needed
+          window.removeEventListener('message', onIFrameCallback);
+        };
+
+        window.addEventListener('message', onIFrameCallback);
+
+        postFrameMessage({
+          captureSourceId: true
+        });
+
+      } else {
+        baseGetUserMedia(constraints, successCb, failureCb);
+      }
+    };
+
+    window.getUserMedia = window.navigator.getUserMedia;
+
+  } else {
+    baseGetUserMedia = window.navigator.getUserMedia;
```

Comment:
In the basic AJS we define
navigator.getUserMedia = getUserMedia;
Therefore I don't think we should refer to window.navigator.getUserMedia here. (Even if it's technically the same)